app/models/template_input.rb

```diff
@@ -91,11 +91,12 @@ def resolved_value
 
   class UserInputResolver < InputResolver
     def ready?
-      !input_value.nil?
+      @renderer.invocation
     end
 
     def resolved_value
-      input_value.value
+      raise(ValueNotReady, _("Required input not specified")) if @input.required? && input_value.try(:value).blank?
```

Comment:
@ares - i can move that check to ready.  

I would think an exception should be thrown if .value is called and a required value is not present.  But i do agree that .ready? should return false.

When you say 'accept blank values', do you mean for required inputs (treating those as provided?)